### ISSUE TITLE
/* treated as comment

### ISSUE THREAD

**Issue by Lunkwill-Fook** [hashicorp/vscode-hcl#86]
/* is treated as comment

e.g.
path "auth/*"
{
  capabilities = ["create", "read", "update", "delete", "list", "sudo"]
}
[...]

everything after /*" is green

**Comment by jpogran** [comment]
Thanks for the bug report @Lunkwill-Fook, we'll figure this out

**Comment by radeksimko** [comment]
@Lunkwill-Fook I tried to reproduce the bug and I wasn't able to, see my rendered example below.

```hcl
# This section grants all access on "secret/*". Further restrictions can be
# applied to this broad policy, as shown below.
path "secret/*" {
  capabilities = ["create", "read", "update", "delete", "list"]
}

path "auth/*" {
  capabilities = ["create", "read", "update", "delete", "list"]
}

```

![Screenshot 2022-05-20 at 13 41 06](https://user-images.githubusercontent.com/287584/169530365-f41b9fab-942b-406b-8f11-1c673ff551e0.png)

Do you mind providing any more details?

1. can you confirm you're on `v0.1.0` of the HCL extension?
1. enable the token/scope inspector via Command Palette (Cmd/Ctrl+Shift+P) and click on the mis-highlighted part of the config 
![Screenshot 2022-05-20 at 13 44 51](https://user-images.githubusercontent.com/287584/169530940-b60b30b7-dc72-4861-97bd-4e1e0437cd55.png) 
![Screenshot 2022-05-20 at 13 47 05](https://user-images.githubusercontent.com/287584/169531080-a006ec47-c3f4-4028-9bd5-73ecde02da23.png)
1. provide a list of all extensions so we can verify there is no other extension claiming this config file? You can run `code --list-extensions` in your terminal.
1. provide any relevant settings (e.g. anything that touches `hcl` language) 
![Screenshot 2022-05-20 at 13 52 00](https://user-images.githubusercontent.com/287584/169531858-e72afc7c-1a4a-43b2-a8a3-3ad28d9ac84c.png)

Thanks.

**Comment by radeksimko** [comment]
We will close this issue as not reproducible in the next 14 days unless we receive further details to allow us reproduce the problem. Thanks for understanding.

**Comment by Lunkwill-Fook** [comment]
Thanks for the feedback.
With the updated version installed from VSIX it works fine!

**Comment by jpogran** [comment]
Great to hear!

**Comment by slumbi** [comment]
Still happens to me with v0.4.0 with a policy like :

```hcl
  path "secrets/data/users/{{identity.entity.name}}/*" {
    capabilities = ["create", "update", "patch", "read", "delete", "list"]
  }
```

**Comment by dbanck** [comment]
@slumbi I can reproduce the behavior you're seeing:

<img width="688" alt="CleanShot 2024-04-19 at 10 33 55@2x" src="https://github.com/hashicorp/vscode-hcl/assets/45985/9a40d8b2-b7cc-4ec9-83cc-29bc1614925f">

Before I open a new issues for this, I want to make sure that I fully understand the context here. I haven't seen `{{` (or `}}`) being used in an hcl-based language before. When I remove this part from the label, everything looks as expected:

<img width="678" alt="CleanShot 2024-04-19 at 10 36 31@2x" src="https://github.com/hashicorp/vscode-hcl/assets/45985/223d2bf1-9ae1-4f2d-b314-11c638bd5795">

Can you share more information about your configuration and what's the origin of `{{identity.entity.name}}`?

**Comment by slumbi** [comment]
This is a working Hashicorp Vault policy , which allows setting dynamic rules based on user metadata. (for example this policy allows  every user full access to secrets under their username )

But In my humble opinion ,multi-line comment start or end tokens should be ignored in any quoted strings.

**Comment by dbanck** [comment]
Appreciate the additional context. I've created https://github.com/hashicorp/syntax/issues/119 to track this bug

**Comment by slumbi** [comment]
Thanks for this. However the main problem was not really the curly text highlight, but instead  the `/*`  after curly braces within s string is considered as block comment-start, and the rest of the file becomes commented out .

**Comment by dbanck** [comment]
@slumbi We have just released a new version [0.5.0](https://github.com/hashicorp/vscode-hcl/releases/tag/v0.5.0). This updates the grammar to the latest version and should fix the highlighting bug. The update should show up automatically in VS Code.

--- 

If you find any other syntax related bugs, please do let us know via a new issue.
